lib/smack/SmackRep.cpp

```diff
@@ -820,7 +820,11 @@ const Expr* SmackRep::cast(const llvm::ConstantExpr* CE) {
 }
 
 const Expr* SmackRep::cast(unsigned opcode, const llvm::Value* v, const llvm::Type* t) {
-  return Expr::fn(opName(Naming::INSTRUCTION_TABLE.at(opcode), {v->getType(), t}), expr(v));
+  std::string fn = Naming::INSTRUCTION_TABLE.at(opcode);
+  if (fn == "$fptrunc" || fn == "$fpext" || fn == "$si2fp") {
```

Comment:
Can't you use `opcode` here instead of `fn`?

Reply:
Meaning, use `Instruction::FPTrunc` instead of `$fptrunc`.

Reply:
How come `Instruction::FPToUI` is not in this if condition?

Reply:
I initally did have fp2si, fp2ui, and ui2fp in this if condition. However, I realized that these functions do not have any builtin functions equivalent to them, according to the [SMT-LIB standard](http://smtlib.cs.uiowa.edu/theories-FloatingPoint.shtml). Instead, these are modeled using uninterpreted functions, as seen [here](https://github.com/smackers/smack/pull/327/files#diff-85650691a95426ad3aec3acc90105837R878), [here](https://github.com/smackers/smack/pull/327/files#diff-85650691a95426ad3aec3acc90105837R977), and [here](https://github.com/smackers/smack/pull/327/files#diff-85650691a95426ad3aec3acc90105837L981) in smack.c.